renku/models/provenance/entities.py

```diff
@@ -59,7 +59,9 @@ def default_id(self):
     @_label.default
     def default_label(self):
         """Generate a default label."""
-        return '{self.path}@{self.commit.hexsha}'.format(self=self)
+        if self.path:
+            return '{self.path}@{self.commit.hexsha}'.format(self=self)
+        return '{self.commit.hexsha}'.format(self=self)
```

Comment:
I guess this will produce different results from what might be in the triple store already? We should probably remove and re- index the dev Jena?

Reply:
Yes, we will have more issues with projects (see #403)